// ngx-social-button/esm2015/lib/entities/baseLoginProviderImpl.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
/**
 * @abstract
 */
export class BaseLoginProviderImpl {
    constructor() {
    }

    /**
     * @param {?} obj
     * @param {?} onload
     * @return {?}
     */
    loadScript(obj, onload) {
        if (document.getElementById(obj.name)) {
            return;
        }
        /** @type {?} */
        let signInJS = document.createElement('script');
        signInJS.async = true;
        signInJS.src = obj.url;
        signInJS.onload = onload;
        if (obj.name === 'linkedin') {
            signInJS.async = false;
            signInJS.text = ('api_key: ' + obj.id).replace('\'', '');
        }
        document.head.appendChild(signInJS);
    }
}

if (false) {
    /** @type {?} */
    BaseLoginProviderImpl.prototype.TYPE;
    /**
     * @abstract
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.initialize = function () {
    };
    /**
     * @abstract
     * @param {?=} scopes
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.signIn = function (scopes) {
    };
    /**
     * @abstract
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.signOut = function () {
    };
    /**
     * @abstract
     * @param {?=} share
     * @return {?}
     */
    BaseLoginProviderImpl.prototype.sharing = function (share) {
    };
}
