// ngx-social-button/esm5/lib/buttons/linkedinLoginButton.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import {Component, EventEmitter, Input, Output} from '@angular/core';
import {SocialService} from '../social.service';

var LinkedinLoginButton = /** @class */ (function () {
    function LinkedinLoginButton(socialAuthService) {
        this.socialAuthService = socialAuthService;
        this.socialUser = new EventEmitter();
    }

    /**
     * @return {?}
     */
    LinkedinLoginButton.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
    };
    /**
     * @return {?}
     */
    LinkedinLoginButton.prototype.socialSignIn = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.socialAuthService.signIn('linkedin', this.scopes).then(function (userData) {
            _this.socialUser.emit(userData);
        }).catch(function () {
            _this.socialUser.emit(null);
        });
    };
    LinkedinLoginButton.decorators = [
        {
            type: Component, args: [{
                selector: 'linkedin-login-button',
                template: "\n\t\t<button [className]=\"style == 'round' ? 'social-btn linkedin' : 'button btn-block loginBtn --linkedin'\"\n\t\t        (click)=\"socialSignIn()\" type=\"button\">\n\t\t\t{{style == 'round' ? '':  text ? text : 'Login with LinkedIn'}}\n\t\t</button>\n    ",
                styles: [".button{font-family:Lato,\"Lucida Grande\",\"Lucida Sans Unicode\",Tahoma,Sans-Serif;-webkit-appearance:none;display:inline-block;outline:0;border-radius:3px;border:3px solid transparent;cursor:pointer;text-overflow:ellipsis;text-decoration:none!important;font-weight:400!important}.loginBtn{box-sizing:border-box;position:relative;margin:.2em;padding:0 15px 0 46px;border:none;text-align:left;line-height:34px;white-space:nowrap;border-radius:3px;font-size:16px;color:#fff}.loginBtn:before{content:\"\";box-sizing:border-box;position:absolute;top:0;left:0;width:34px;height:100%}.loginBtn:focus{outline:0}.loginBtn:active{box-shadow:inset 0 0 0 32px rgba(0,0,0,.1)}.shareBtn{box-sizing:border-box;position:relative;margin:.2em;padding:0 15px 0 46px;border:none;text-align:left;line-height:34px;white-space:nowrap;border-radius:3px;font-size:16px;color:#fff}.shareBtn:before{content:\"\";box-sizing:border-box;position:absolute;top:0;left:0;width:34px;height:100%}.shareBtn:focus{outline:0}.shareBtn:active{box-shadow:inset 0 0 0 32px rgba(0,0,0,.1)}.--facebook{background-color:#4c69ba;background-image:linear-gradient(#4c69ba,#3b55a0);text-shadow:0 -1px 0 #354c8c}.--facebook:before{border-right:1px solid #364e92;background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/14082/icon_facebook.png) 6px 6px no-repeat}.--facebook:focus,.--facebook:hover{background-color:#5b7bd5;background-image:linear-gradient(#5b7bd5,#4864b1)}.--google{background:#dd4b39}.--google:before{border-right:1px solid #bb3f30;background:url(https://s3-us-west-2.amazonaws.com/s.cdpn.io/14082/icon_google.png) 6px 6px no-repeat}.--google:focus,.--google:hover{background:#e74b37}.--linkedin{background:#007bb6}.--linkedin:before{border-right:1px solid #007bb6;background:url(https://content.linkedin.com/content/dam/developer/global/en_US/site/img/signin-button.png) -3px -3px no-repeat}.--linkedin:focus,.--linkedin:hover{background:linear-gradient(#007bb6,#4864b1)}.social-btn{width:50px;height:50px;background-size:cover;background-repeat:no-repeat;cursor:pointer;border-radius:10px;border:none}.social-btn.facebook{background-image:url(https://raw.githubusercontent.com/JohnnieLi/ngx-social-button/master/assets/facebook.png)}.social-btn.google{background-image:url(https://raw.githubusercontent.com/JohnnieLi/ngx-social-button/master/assets/google.png)}.social-btn.linkedin{background-image:url(https://raw.githubusercontent.com/JohnnieLi/ngx-social-button/master/assets/LinkedIn.png)}.social-btn.wechat{background-image:url(https://github.com/JohnnieLi/ngx-social-button/blob/master/assets/wechat.png?raw=true)}"],
            },]
        },
    ];
    /** @nocollapse */
    LinkedinLoginButton.ctorParameters = function () {
        return [
            {type: SocialService}
        ];
    };
    LinkedinLoginButton.propDecorators = {
        size: [{type: Input}],
        style: [{type: Input}],
        text: [{type: Input}],
        scopes: [{type: Input}],
        socialUser: [{type: Output}]
    };
    return LinkedinLoginButton;
}());
export {LinkedinLoginButton};
if (false) {
    /** @type {?} */
    LinkedinLoginButton.prototype.size;
    /** @type {?} */
    LinkedinLoginButton.prototype.style;
    /** @type {?} */
    LinkedinLoginButton.prototype.text;
    /** @type {?} */
    LinkedinLoginButton.prototype.scopes;
    /** @type {?} */
    LinkedinLoginButton.prototype.socialUser;
    /** @type {?} */
    LinkedinLoginButton.prototype.socialAuthService;
}
